clippy_lints/src/operators/arithmetic_side_effects.rs

```diff
@@ -122,6 +123,9 @@ impl ArithmeticSideEffects {
         lhs: &hir::Expr<'tcx>,
         rhs: &hir::Expr<'tcx>,
     ) {
+        if is_from_proc_macro(cx, expr) {
+            return;
+        }
```

Comment:
Can this check be moved to `fn check_expr`? The same can apply to the check in `fn manage_unary_ops`.

Reply:
Moved to `should_skip_expr`